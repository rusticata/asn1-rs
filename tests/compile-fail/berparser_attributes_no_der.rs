use asn1_rs::*;
use hex_literal::hex;

fn derive_choice_explicit_attributes_no_der() {
    #[derive(Debug, PartialEq)]
    //
    #[derive(Choice)]
    #[tagged_explicit]
    // generate parsers only
    #[asn1(parse = "BER")]
    pub enum MyEnum {
        Val0(u8),
        Val1(String),
        Val2(u32),
        Val3(Vec<u32>),
    }

    //--- no DER parser
    let bytes = &hex!("a0 03 020108");
    // parse back
    let _ = MyEnum::parse_ber(Input::from(Input::from(bytes))).expect("parsing BER failed");
    let _ = MyEnum::parse_der(Input::from(Input::from(bytes))).expect("parsing DER failed");
}

fn main() {
    derive_choice_explicit_attributes_no_der();
}
