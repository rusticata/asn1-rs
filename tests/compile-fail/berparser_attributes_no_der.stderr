error[E0599]: no variant or associated item named `parse_der` found for enum `MyEnum` in the current scope
  --> tests/compile-fail/berparser_attributes_no_der.rs:22:21
   |
11 |     pub enum MyEnum {
   |     --------------- variant or associated item `parse_der` not found for this enum
...
22 |     let _ = MyEnum::parse_der(Input::from(Input::from(bytes))).expect("parsing DER failed");
   |                     ^^^^^^^^^ variant or associated item not found in `MyEnum`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `parse_der`, perhaps you need to implement it:
           candidate #1: `derive_choice_explicit_attributes_no_der::_::asn1_rs::DerParser`
help: there is an associated function `parse_ber` with a similar name
   |
22 -     let _ = MyEnum::parse_der(Input::from(Input::from(bytes))).expect("parsing DER failed");
22 +     let _ = MyEnum::parse_ber(Input::from(Input::from(bytes))).expect("parsing DER failed");
   |
